packages/twenty-server/src/modules/workflow/workflow-status/jobs/workflow-statuses-update.job.ts

```
@@ -170,17 +170,23 @@ export class WorkflowStatusesUpdateJob {
                 workspaceId,
               );
           } catch (e) {
-            serverlessFunction = null;
+            serverlessFunction = (
+              await this.serverlessFunctionService.findManyServerlessFunctions({
+                id: step.settings.input.serverlessFunctionId,
+              })
+            )?.[0];
           }
 
-          if (serverlessFunction) {
-            const newStepSettings = { ...step.settings };
+          if (!serverlessFunction) {
+            throw new Error('Published serverless function does not exist');
```

can we throw a specific exception here 🙏